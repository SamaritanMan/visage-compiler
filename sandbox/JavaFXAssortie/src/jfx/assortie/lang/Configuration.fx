package jfx.assortie.lang;

import jfx.assortie.system.Module;
import jfx.assortie.system.Sample;

Module {
    name: "JavaFX"
    modules: [
    "jfx.assortie.lang.api.Configuration",
    ]
    samples: [
        Sample{
            name: "New"
            className: "jfx.assortie.samples.empty.Empty"
            visible: true
        }
    ]

}
