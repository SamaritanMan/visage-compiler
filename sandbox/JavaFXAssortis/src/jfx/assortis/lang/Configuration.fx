package jfx.assortis.lang;

import jfx.assortis.system.Module;
import jfx.assortis.system.Sample;

Module {
    name: "JavaFX"
    modules: [
    "jfx.assortis.lang.api.Configuration",
    "jfx.assortis.lang.animation.Configuration",
    ]
    samples: [
        Sample{
            name: "New"
            className: "jfx.assortis.samples.empty.Empty"
            visible: true
        }
    ]

}
