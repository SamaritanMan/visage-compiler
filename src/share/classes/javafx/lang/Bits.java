package javafx.lang;

/**
 *  Bit operations
 */

public class Bits {

    public static int shiftLeft(int value, int shift) {
        return value << shift;
    }

    public static int shiftRight(int value, int shift) {
        return value >> shift;
    }

    public static int bitOr(int value, int mask) {
        return value | mask;
    }

    public static int bitAnd(int value, int mask) {
        return value & mask;
    }
    
    public static int bitXor(int value, int mask) {
        return value ^ mask;
    }

    public static int complement(int value) {
        return ~value;
    }

    public static int add(int value, int mask) {
        return value | mask;
    }

    public static int remove(int value, int mask) {
        return value & ~mask;
    }

    public static boolean contains(int value, int mask) {
        return (value & mask) != 0;
    }


    public static long shiftLeft(long value, int shift) {
        return value << shift;
    }

    public static long shiftRight(long value, int shift) {
        return value >> shift;
    }

    public static long bitOr(long value, long mask) {
        return value | mask;
    }

    public static long bitAnd(long value, long mask) {
        return value & mask;
    }
    
    public static long bitXor(long value, long mask) {
        return value ^ mask;
    }

    public static long complement(long value) {
        return ~value ;
    }

    public static long add(long value, long mask) {
        return value | mask;
    }

    public static long remove(long value, long mask) {
        return value & ~mask;
    }

    public static boolean contains(long value, long mask) {
        return (value & mask) != 0;
    }

}
