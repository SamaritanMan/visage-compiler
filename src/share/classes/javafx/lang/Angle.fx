/*
 * Copyright 2008-2009 Sun Microsystems, Inc.  All Rights Reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Sun Microsystems, Inc., 4150 Network Circle, Santa Clara,
 * CA 95054 USA or visit www.sun.com if you need additional information or
 * have any questions.
 */

package javafx.lang;

import java.lang.Comparable;
import java.lang.IllegalStateException;
import java.lang.Math;
import java.lang.ThreadLocal;
import java.text.DecimalFormat;
import java.util.Formattable;
import javafx.util.Bits;
import java.util.Formatter;
import java.util.FormattableFlags;
import java.lang.StringBuilder;

/**
 * Used to specify a angle of 0ms.
 *
 * @profile common
 */
public def ZERO:Angle = Angle {value: 0, unit: AngleUnit.RADIAN}

def DF = ThreadLocal {
    override function initialValue():Object {
        return new DecimalFormat("#.######");
    }
}

// script-level "static" functions below

/**
 * Factory method that returns a Angle instance for a specified
 * value and unit.
 *
 * @param value the magnitude of the angle
 * @param unit the unit of the angle
 * @return a Angle instance of the specified value and unit type
 * @profile common
 */
public function valueOf(value:Double, unit:AngleUnit):Angle {
    if (value == 0)
        ZERO
    else Angle {
        value: value
        unit: unit
    }
}

public function valueOf(value:Float, unit:AngleUnit):Angle {
    valueOf(value as Double, unit)
}

public def TYPE_INFO = com.sun.javafx.runtime.TypeInfo.makeAndRegisterTypeInfo(ZERO);

/**
 * A class that defines an angle measurement.  Angle instances can be specified
 * as degrees, radians, or turns.  For precise calculations, they are stored
 * in the given unit type as a double value.
 * <p>
 * Angles can be easily created using angle literals; for
 * example, a 90 degree Angle instance can be defined in several
 * ways:
 * <code><pre>    Angle a = 90deg;
    Angle a = .25turn;
    Angle a = Angle.valueOf(Math.PI / 2, AngleUnit.RADIAN);</pre></code>
 * Angle instances are immutable, and are therefore replaced rather than modified.
 * To create a new Angle instance, either use a time literal, or use the
 * <code>Angle.valueOf(value, unit)</code> factory method.
 *
 * @profile common
 */
public class Angle extends Comparable, Formattable {

    var value:Double;
    var unit:AngleUnit;

    /**
     * Returns the number of degrees in this angle.
     *
     * @profile common
     */
    public function toDegrees():Double {
        if (unit == AngleUnit.DEGREE) {
            return value;
        } else if (unit == AngleUnit.RADIAN) {
            return value * 180 / Math.PI;
        } else if (unit == AngleUnit.TURN) {
            return value * 360;
        } else {
            throw new IllegalStateException("Unknown unit: {unit}");
        }
    }

    /**
     * Returns a new Length object with an angle in degrees.
     *
     * @profile common
     */
    public function toDegreeAngle():Angle {
        if (this == ZERO) this;
        Angle {
            value: toDegrees();
            unit: AngleUnit.DEGREE;
        }
    }
    
    /**
     * Returns the number of radians in this angle.
     *
     * @profile common
     */
    public function toRadians():Number {
        if (unit == AngleUnit.RADIAN) {
            return value;
        } else if (unit == AngleUnit.DEGREE) {
            return value * Math.PI / 180;
        } else if (unit == AngleUnit.TURN) {
            return value * 2 * Math.PI;
        } else {
            throw new IllegalStateException("Unknown unit: {unit}");
        }
    }
    
    /**
     * Returns a new Length object with an angle in radians.
     *
     * @profile common
     */
    public function toRadianAngle():Angle {
        if (this == ZERO) this;
        Angle {
            value: toRadians();
            unit: AngleUnit.RADIAN;
        }
    }

    /**
     * Returns the number of turns in this angle.
     *
     * @profile common
     */
    public function toTurns():Number {
        if (unit == AngleUnit.TURN) {
            return value;
        } else if (unit == AngleUnit.DEGREE) {
            return value / 360;
        } else if (unit == AngleUnit.RADIAN) {
            return value / 2 / Math.PI;
        } else {
            throw new IllegalStateException("Unknown unit: {unit}");
        }
    }

    /**
     * Returns a new Length object with an angle in turns.
     *
     * @profile common
     */
    public function toTurnAngle():Angle {
        if (this == ZERO) this;
        Angle {
            value: toTurns();
            unit: AngleUnit.TURN;
        }
    }

    /**
     * Add this instance and another Angle instance to return a new Angle instance.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */
    public function add(other:Angle):Angle {
        if (unit == other.unit) {
            return valueOf(value + other.value, unit);
        } else {
            return valueOf(toRadians() + other.toRadians(), AngleUnit.RADIAN);
        }
    }

    /**
     * Subtract other Angle instance from this instance to return a new Angle instance.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */
    public function sub(other:Angle):Angle {
        if (unit == other.unit) {
            return valueOf(value - other.value, unit);
        } else {
            return valueOf(toRadians() - other.toRadians(), AngleUnit.RADIAN);
        }
    }

    /**
     * Multiply this instance with a number to return a new Angle instance.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */ 
    public function mul(n:Number):Angle {
        return valueOf(value * n, unit);
    }


    /**
     * Divide this instance by a number to return a new Angle instance.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */     
    public function div(n:Number):Angle {
        if (n == 0) {
            throw new java.lang.ArithmeticException("/ by zero");
        }
        
        return valueOf(value / n, unit);
    }

    /**
     * Divide this instance by another Angle to return the ratio.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */     
    public function div(other:Angle):Double {
        if (other.value == 0) {
            throw new java.lang.ArithmeticException("/ by zero");
        }
        if (unit == other.unit) {
            return value / other.value;
        } else {
            return toRadians() / other.toRadians();
        }
    }

    /** 
     * Return a new Angle instance which has a negative number of milliseconds
     * from this instance.  For example, <code>(50deg).negate()</code> returns
     * a Angle of -50 degrees.
     * This function does not change the value of the called Angle instance.
     *
     * @profile common
     */
    public function negate():Angle {
        return valueOf(-value, unit);
    }

    /**
     * Prints out a string representation of the length with a precision of
     * up to 6 digits (internally angles are represented as doubles with much
     * higher precision).
     *
     * @profile common
     */        
    override function toString():String {
        def df = DF.get() as DecimalFormat;
        df.setMinimumFractionDigits(0);
        return "{df.format(value)}{unit.getSuffix()}";
    }

    /**
     * Formatting function that allows angles to be properly formatted by
     * printf, String.format, or the JavaFX format primitive.
     * <p>
     * Width sets the minimum width of this field, following the justification
     * rules set in flags.
     * <p>
     * Precision can be used to specify exactly how many digits should appear
     * to the right of the decimal, including zero padding.
     * <p>
     * The alternate form is not supported for angles.
     * <p>
     * Finally, the uppercase flag is also supported, but only applies to the
     * length suffix.
     */
    override function formatTo(formatter:Formatter, flags:Integer, width:Integer, precision:Integer):Void {
        def uppercase = Bits.bitAnd(flags, FormattableFlags.UPPERCASE) != 0;
        def leftJustify = Bits.bitAnd(flags, FormattableFlags.LEFT_JUSTIFY) != 0;
        def sb = StringBuilder {};
        def df = DF.get() as DecimalFormat;
        df.setMinimumFractionDigits(precision);
        sb.append(df.format(value));
        sb.append(if (uppercase) unit.getSuffix().toUpperCase() else unit.getSuffix());
        def len = sb.length();
        if (len < width) {
            if (leftJustify) {
                for (i in [1..width-len]) sb.append(' ');
            } else {
                for (i in [1..width-len]) sb.<<insert>>(0, ' ');
            }
        }
        formatter.out().append(sb);
    }

    /**
     * Returns true if the specified angle is less than (<) this instance.
     * INDEFINITE is treated as if it were positive infinity.
     *
     * @profile common
     */
    public function lt(other:Angle):Boolean {
        return compareTo(other) < 0;
    }

    /** 
     * Returns true if the specified angle is less than or equal to (<=) this instance.
     * INDEFINITE is treated as if it were positive infinity.
     *
     * @profile common
     */
    public function le(other:Angle):Boolean {
        return compareTo(other) <= 0;
    }

    /** 
     * Returns true if the specified angle is greater than (>) this instance.
     * INDEFINITE is treated as if it were positive infinity.
     *
     * @profile common
     */
    public function gt(other:Angle):Boolean {
        return compareTo(other) > 0;
    }

    /**
     * Returns true if the specified angle is greater than or equal to (>=) this instance.
     * INDEFINITE is treated as if it were positive infinity.
     *
     * @profile common
     */
    public function ge(other:Angle):Boolean {
        return compareTo(other) >= 0;
    }

    override function compareTo(obj:Object):Integer {
        def a = obj as Angle;
        if (a.value == value and a.unit == unit) {
            return 0;
        }
        def cmp = toRadians() - a.toRadians();
        return if (cmp < 0)
             -1
        else if(cmp > 0)
             1
        else
             0
    }

    override function equals(obj:Object):Boolean {
        if (obj instanceof Angle) {
            def a = obj as Angle;
            if (isSameObject(a, this)) {
                return true;
            }
            return a.toRadians() == toRadians();
        }
        return false;
    }

    override function hashCode():Integer {
        var v = value as Long;
        v = Bits.bitXor(v, unit.ordinal());
        return Bits.bitXor(v, Bits.unsignedShiftRight(v, 32)) as Integer;
    }
}
