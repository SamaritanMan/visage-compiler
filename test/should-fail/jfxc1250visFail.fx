/**
 * Regression test JFXC-1250 : visibility modifiers
 *
 * @compilefirst jfxc1250/a/jfxc1250subFailA.fx
 * @compilefirst jfxc1250subFail.fx
 * @test/fail
 */
import java.lang.System;
import jfxc1250.a.jfxc1250subFailA;

public var slPub = 3;
package var slPkg = 3;
var slScr = 3;
non-writable public var slNPub = 3;
non-writable package var slNPkg = 3;
non-writable var slNScr = 3;
public-readable package var slRPkg = 3;
public-readable var slRScr = 3;
non-writable public-readable package var slNRPkg = 3;
non-writable public-readable var slNRScr = 3;

public function slFPub() { 3 }
package function slFPkg() { 3 }
function slFScr() { 3 }

public class cPub {}
package class cPkg {}
class cScr {}

public class jfxc1250visFail {
	public var memPub = 3;
	protected var memPro = 3;
	package var memPkg = 3;
	var memScr = 3;
	non-writable public var memNPub = 3;
	non-writable protected var memNPro = 3;
	non-writable package var memNPkg = 3;
	non-writable var memNScr = 3;
	public-readable protected var memRPro = 3;
	public-readable package var memRPkg = 3;
	public-readable var memRScr = 3;
	non-writable public-readable protected var memNRPro = 3;
	non-writable public-readable package var memNRPkg = 3;
	non-writable public-readable var memNRScr = 3;

	public function memFPub() { 3 }
	protected function memFPro() { 3 }
	package function memFPkg() { 3 }
	function memFScr() { 3 }
}

class Foo {
	public var fooPub = 3;
	protected var fooPro = 3;
	package var fooPkg = 3;
	var fooScr = 3;
	non-writable public var fooNPub = 3;
	non-writable protected var fooNPro = 3;
	non-writable package var fooNPkg = 3;
	non-writable var fooNScr = 3;
	public-readable protected var fooRPro = 3;
	public-readable package var fooRPkg = 3;
	public-readable var fooRScr = 3;
	non-writable public-readable protected var fooNRPro = 3;
	non-writable public-readable package var fooNRPkg = 3;
	non-writable public-readable var fooNRScr = 3;

	public function fooFPub() { 3 }
	protected function fooFPro() { 3 }
	package function fooFPkg() { 3 }
	function fooFScr() { 3 }
}

function slVisRead() {
	System.out.println(jfxc1250visFail.slPub);
	System.out.println(jfxc1250visFail.slPkg);
	System.out.println(jfxc1250visFail.slScr);
	System.out.println(jfxc1250visFail.slNPub);
	System.out.println(jfxc1250visFail.slNPkg);
	System.out.println(jfxc1250visFail.slNScr);
	System.out.println(jfxc1250visFail.slRPkg);
	System.out.println(jfxc1250visFail.slRScr);
	System.out.println(jfxc1250visFail.slNRPkg);
	System.out.println(jfxc1250visFail.slNRScr);
}

function memVisRead() {
	var vf = jfxc1250visFail {};
	System.out.println(vf.memPub);
	System.out.println(vf.memPkg);
	System.out.println(vf.memPro);
	System.out.println(vf.memScr);
	System.out.println(vf.memNPub);
	System.out.println(vf.memNPkg);
	System.out.println(vf.memNPro);
	System.out.println(vf.memNScr);
	System.out.println(vf.memRPro);
	System.out.println(vf.memRPkg);
	System.out.println(vf.memRScr);
	System.out.println(vf.memNRPro);
	System.out.println(vf.memNRPkg);
	System.out.println(vf.memNRScr);
}

function fooVisRead() {
	var fv = Foo {};
	System.out.println(fv.fooPub);
	System.out.println(fv.fooPkg);
	System.out.println(fv.fooPro);
	System.out.println(fv.fooScr);
	System.out.println(fv.fooNPub);
	System.out.println(fv.fooNPkg);
	System.out.println(fv.fooNPro);
	System.out.println(fv.fooNScr);
	System.out.println(fv.fooRPro);
	System.out.println(fv.fooRPkg);
	System.out.println(fv.fooRScr);
	System.out.println(fv.fooNRPro);
	System.out.println(fv.fooNRPkg);
	System.out.println(fv.fooNRScr);
}

function slVisCall() {
	System.out.println(jfxc1250visFail.slFPub());
	System.out.println(jfxc1250visFail.slFPkg());
	System.out.println(jfxc1250visFail.slFScr());
}

function memVisCall() {
	var vf = jfxc1250visFail {};
	System.out.println(vf.memFPub());
	System.out.println(vf.memFPkg());
	System.out.println(vf.memFScr());
}

function cVisNew() {
	var x1 = jfxc1250visFail.cPub {};
	var x2 = jfxc1250visFail.cPkg {};
	var x3 = jfxc1250visFail.cScr {};
}

// ---

function slSubRead() {
	System.out.println(jfxc1250subFail.slPub);
	System.out.println(jfxc1250subFail.slPkg);
	System.out.println(jfxc1250subFail.slScr);
	System.out.println(jfxc1250subFail.slNPub);
	System.out.println(jfxc1250subFail.slNPkg);
	System.out.println(jfxc1250subFail.slNScr);
	System.out.println(jfxc1250subFail.slRPkg);
	System.out.println(jfxc1250subFail.slRScr);
	System.out.println(jfxc1250subFail.slNRPkg);
	System.out.println(jfxc1250subFail.slNRScr);
}

function memSubRead() {
	var sf = jfxc1250subFail {};
	System.out.println(sf.memPub);
	System.out.println(sf.memPkg);
	System.out.println(sf.memPro);
	System.out.println(sf.memScr);
	System.out.println(sf.memNPub);
	System.out.println(sf.memNPkg);
	System.out.println(sf.memNPro);
	System.out.println(sf.memNScr);
	System.out.println(sf.memRPro);
	System.out.println(sf.memRPkg);
	System.out.println(sf.memRScr);
	System.out.println(sf.memNRPro);
	System.out.println(sf.memNRPkg);
	System.out.println(sf.memNRScr);
}

function slSubCall() {
	System.out.println(jfxc1250subFail.slFPub());
	System.out.println(jfxc1250subFail.slFPkg());
	System.out.println(jfxc1250subFail.slFScr());
}

function memSubCall() {
	var sf = jfxc1250subFail {};
	System.out.println(sf.memFPub());
	System.out.println(sf.memFPkg());
	System.out.println(sf.memFScr());
}

function cSubNew() {
	var x1 = jfxc1250subFail.cPub {};
	var x2 = jfxc1250subFail.cPkg {};
	var x3 = jfxc1250subFail.cScr {};
}

//---

function slARead() {
	System.out.println(jfxc1250subFailA.slPub);
	System.out.println(jfxc1250subFailA.slPkg);
	System.out.println(jfxc1250subFailA.slScr);
	System.out.println(jfxc1250subFailA.slNPub);
	System.out.println(jfxc1250subFailA.slNPkg);
	System.out.println(jfxc1250subFailA.slNScr);
	System.out.println(jfxc1250subFailA.slRPkg);
	System.out.println(jfxc1250subFailA.slRScr);
	System.out.println(jfxc1250subFailA.slNRPkg);
	System.out.println(jfxc1250subFailA.slNRScr);
}

function memARead() {
	var sfa = jfxc1250subFailA {};
	System.out.println(sfa.memPub);
	System.out.println(sfa.memPkg);
	System.out.println(sfa.memPro);
	System.out.println(sfa.memScr);
	System.out.println(sfa.memNPub);
	System.out.println(sfa.memNPkg);
	System.out.println(sfa.memNPro);
	System.out.println(sfa.memNScr);
	System.out.println(sfa.memRPro);
	System.out.println(sfa.memRPkg);
	System.out.println(sfa.memRScr);
	System.out.println(sfa.memNRPro);
	System.out.println(sfa.memNRPkg);
	System.out.println(sfa.memNRScr);
}

function slACall() {
	System.out.println(jfxc1250subFailA.slFPub());
	System.out.println(jfxc1250subFailA.slFPkg());
	System.out.println(jfxc1250subFailA.slFScr());
}

function memACall() {
	var sfa = jfxc1250subFailA {};
	System.out.println(sfa.memFPub());
	System.out.println(sfa.memFPkg());
	System.out.println(sfa.memFScr());
}

function cANew() {
	var x1 = jfxc1250subFailA.cPub {};
	var x2 = jfxc1250subFailA.cPkg {};
	var x3 = jfxc1250subFailA.cScr {};
}


function run( ) {
	slVisRead();
	memVisRead();
	fooVisRead();
	slVisCall();
	memVisCall();
	cVisNew();

	slSubRead();
	memSubRead();
	slSubCall();
	memSubCall();
	cSubNew();

	slARead();
	memARead();
	slACall();
	memACall();
	cANew();
}