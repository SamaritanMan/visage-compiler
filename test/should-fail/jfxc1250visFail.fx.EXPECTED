test\should-fail\jfxc1250visFail.fx:147: slScr has script only (default) access in jfxc1250subFail
	System.out.println(jfxc1250subFail.slScr);
	                                  ^
test\should-fail\jfxc1250visFail.fx:150: slNScr has script only (default) access in jfxc1250subFail
	System.out.println(jfxc1250subFail.slNScr);
	                                  ^
test\should-fail\jfxc1250visFail.fx:162: memScr has script only (default) access in jfxc1250subFail
	System.out.println(sf.memScr);
	                     ^
test\should-fail\jfxc1250visFail.fx:166: memNScr has script only (default) access in jfxc1250subFail
	System.out.println(sf.memNScr);
	                     ^
test\should-fail\jfxc1250visFail.fx:178: slFScr() has script only (default) access in jfxc1250subFail
	System.out.println(jfxc1250subFail.slFScr());
	                                  ^
test\should-fail\jfxc1250visFail.fx:185: memFScr() has script only (default) access in jfxc1250subFail
	System.out.println(sf.memFScr());
	                     ^
test\should-fail\jfxc1250visFail.fx:191: jfxc1250subFail.cScr has script only (default) access in jfxc1250subFail
	var x3 = jfxc1250subFail.cScr {};
	                        ^
test\should-fail\jfxc1250visFail.fx:198: slPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(jfxc1250subFailA.slPkg);
	                                   ^
test\should-fail\jfxc1250visFail.fx:199: slScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(jfxc1250subFailA.slScr);
	                                   ^
test\should-fail\jfxc1250visFail.fx:201: slNPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(jfxc1250subFailA.slNPkg);
	                                   ^
test\should-fail\jfxc1250visFail.fx:202: slNScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(jfxc1250subFailA.slNScr);
	                                   ^
test\should-fail\jfxc1250visFail.fx:212: memPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(sfa.memPkg);
	                      ^
test\should-fail\jfxc1250visFail.fx:213: memPro has protected access in jfxc1250.a.jfxc1250subFailA
	System.out.println(sfa.memPro);
	                      ^
test\should-fail\jfxc1250visFail.fx:214: memScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(sfa.memScr);
	                      ^
test\should-fail\jfxc1250visFail.fx:216: memNPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(sfa.memNPkg);
	                      ^
test\should-fail\jfxc1250visFail.fx:217: memNPro has protected access in jfxc1250.a.jfxc1250subFailA
	System.out.println(sfa.memNPro);
	                      ^
test\should-fail\jfxc1250visFail.fx:218: memNScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(sfa.memNScr);
	                      ^
test\should-fail\jfxc1250visFail.fx:229: slFPkg() is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(jfxc1250subFailA.slFPkg());
	                                   ^
test\should-fail\jfxc1250visFail.fx:230: slFScr() has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(jfxc1250subFailA.slFScr());
	                                   ^
test\should-fail\jfxc1250visFail.fx:236: memFPkg() is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	System.out.println(sfa.memFPkg());
	                      ^
test\should-fail\jfxc1250visFail.fx:237: memFScr() has script only (default) access in jfxc1250.a.jfxc1250subFailA
	System.out.println(sfa.memFScr());
	                      ^
test\should-fail\jfxc1250visFail.fx:242: jfxc1250.a.jfxc1250subFailA.cPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	var x2 = jfxc1250subFailA.cPkg {};
	                         ^
test\should-fail\jfxc1250visFail.fx:243: jfxc1250.a.jfxc1250subFailA.cScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	var x3 = jfxc1250subFailA.cScr {};
	                         ^
test\should-fail\jfxc1250visFail.fx:252: Cannot assign to non-writable 'slNPub'
	jfxc1250visFail.slNPub = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:253: Cannot assign to non-writable 'slNPkg'
	jfxc1250visFail.slNPkg = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:254: Cannot assign to non-writable 'slNScr'
	jfxc1250visFail.slNScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:257: Cannot assign to non-writable 'slNRPkg'
	jfxc1250visFail.slNRPkg = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:258: Cannot assign to non-writable 'slNRScr'
	jfxc1250visFail.slNRScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:267: Cannot assign to non-writable 'memNPub'
	vf.memNPub = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:268: Cannot assign to non-writable 'memNPkg'
	vf.memNPkg = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:269: Cannot assign to non-writable 'memNPro'
	vf.memNPro = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:270: Cannot assign to non-writable 'memNScr'
	vf.memNScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:274: Cannot assign to non-writable 'memNRPro'
	vf.memNRPro = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:275: Cannot assign to non-writable 'memNRPkg'
	vf.memNRPkg = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:276: Cannot assign to non-writable 'memNRScr'
	vf.memNRScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:285: Cannot assign to non-writable 'fooNPub'
	fv.fooNPub = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:286: Cannot assign to non-writable 'fooNPkg'
	fv.fooNPkg = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:287: Cannot assign to non-writable 'fooNPro'
	fv.fooNPro = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:288: Cannot assign to non-writable 'fooNScr'
	fv.fooNScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:292: Cannot assign to non-writable 'fooNRPro'
	fv.fooNRPro = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:293: Cannot assign to non-writable 'fooNRPkg'
	fv.fooNRPkg = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:294: Cannot assign to non-writable 'fooNRScr'
	fv.fooNRScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:300: slScr has script only (default) access in jfxc1250subFail
	jfxc1250subFail.slScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:303: slNScr has script only (default) access in jfxc1250subFail
	jfxc1250subFail.slNScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:305: slRScr has script only (default) write access in jfxc1250subFail
	jfxc1250subFail.slRScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:307: slNRScr has script only (default) write access in jfxc1250subFail
	jfxc1250subFail.slNRScr = 99;
	               ^
test\should-fail\jfxc1250visFail.fx:315: memScr has script only (default) access in jfxc1250subFail
	sf.memScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:319: memNScr has script only (default) access in jfxc1250subFail
	sf.memNScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:322: memRScr has script only (default) write access in jfxc1250subFail
	sf.memRScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:325: memNRScr has script only (default) write access in jfxc1250subFail
	sf.memNRScr = 99;
	  ^
test\should-fail\jfxc1250visFail.fx:330: slPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	jfxc1250subFailA.slPkg = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:331: slScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slScr = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:333: slNPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	jfxc1250subFailA.slNPkg = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:334: slNScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slNScr = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:335: slRPkg has package write access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slRPkg = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:336: slRScr has script only (default) write access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slRScr = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:337: slNRPkg has package write access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slNRPkg = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:338: slNRScr has script only (default) write access in jfxc1250.a.jfxc1250subFailA
	jfxc1250subFailA.slNRScr = 99;
	                ^
test\should-fail\jfxc1250visFail.fx:344: memPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	sfa.memPkg = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:345: memPro has protected access in jfxc1250.a.jfxc1250subFailA
	sfa.memPro = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:346: memScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	sfa.memScr = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:348: memNPkg is not public in jfxc1250.a.jfxc1250subFailA; cannot be accessed from outside package
	sfa.memNPkg = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:349: memNPro has protected access in jfxc1250.a.jfxc1250subFailA
	sfa.memNPro = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:350: memNScr has script only (default) access in jfxc1250.a.jfxc1250subFailA
	sfa.memNScr = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:351: memRPro has protected write access in jfxc1250.a.jfxc1250subFailA
	sfa.memRPro = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:352: memRPkg has package write access in jfxc1250.a.jfxc1250subFailA
	sfa.memRPkg = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:353: memRScr has script only (default) write access in jfxc1250.a.jfxc1250subFailA
	sfa.memRScr = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:354: memNRPro has protected write access in jfxc1250.a.jfxc1250subFailA
	sfa.memNRPro = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:355: memNRPkg has package write access in jfxc1250.a.jfxc1250subFailA
	sfa.memNRPkg = 99;
	   ^
test\should-fail\jfxc1250visFail.fx:356: memNRScr has script only (default) write access in jfxc1250.a.jfxc1250subFailA
	sfa.memNRScr = 99;
	   ^
70 errors
